### PR TITLE
Add support for PCIe GEN5 devices

### PR DESCRIPTION
**Pull request by xyan264**
Background:
- PCIe GEN5 devices are full compatible with GEN4 devices at the kernel driver level

This PR:
- Adds GEN5 devices IDs to the list of supported devices
- Adds checking for `SWITCHTEC_GEN5` to where `SWITCHTEC_GEN4` is used

### CODE REVIEW

File: switchtec.c [hunk 1]
```diff
@@ -363,7 +363,7 @@ static ssize_t field ## _show(struct device *dev, \
 	struct switchtec_dev *stdev = to_stdev(dev); \
 	struct sys_info_regs __iomem *si = stdev->mmio_sys_info; \
 	\
-	if (stdev->gen == SWITCHTEC_GEN4) \
+	if (stdev->gen == SWITCHTEC_GEN4 || stdev->gen == SWITCHTEC_GEN5) \
```
Comment (lsgunth) [inline switchtec.c:366]: Should we maybe just make this >= GEN4? Instead of creating a list that needs to be added to next generation?

Comment (kelvin-cao) [reply switchtec.c:366]: I second it. We can use >= GEN4 till there's incompatibility in the hardware so that we must distinguish GEN4, GEN5.

Comment (xyan264) [reply switchtec.c]: Code updated. Please review and approve if nothing else.

**Review by lsgunth — APPROVED** [review]
Looks good to me.

**Review by kelvin-cao — APPROVED** [review]
Thanks.

**Pull request merged**